### PR TITLE
FRDQ-78 - Add protected route functionality

### PR DESCRIPTION
**Pull request by TomaszKorenberg**

### CODE REVIEW

**Comment by hayuna** [comment]
I think your changes are overcomplicated. Lets call and talk about the details

**Comment by hayuna** [comment]
@PiotrGrobelak Could you take a look into this? I don't have much time right now

File: src/components/atoms/PrivateRoute/PrivateRoute.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import React from 'react';
+import { fakeAuth } from 'Routes/helpers';
+
+const PrivateRoute = ({ children, ...rest }: RouteProps): JSX.Element => {
+  return (
+    <Route
+      {...rest}
+      render={() => (fakeAuth.isAuthenticated ? children : <Redirect to="/login" />)}
+    />
+  );
```
Comment (hayuna) [inline src/components/atoms/PrivateRoute/PrivateRoute.tsx:11]: Check if this is working in the same way
```suggestion
fakeAuth.isAuthenticated && return <Route {...rest}>{children}</Route>
return <Redirect to="/login" />
```

File: src/Routes/helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    this.isAuthenticated = true;
+    setTimeout(cb, 1000);
+  },
+};
+
+export const LoginPage = (): JSX.Element => {
+  const history = useHistory();
+  const location = useLocation();
+  // remember to do this at login page. If you click the back button at this point you don't go back to login page
+  const { from }: any = location.state || { from: { pathname: '/' } };
```
Comment (hayuna) [inline src/Routes/helpers.tsx:17]: Please use better type

File: src/components/atoms/PrivateRoute/PrivateRoute.tsx (hunk 1 above)
Comment (TomaszKorenberg) [reply src/components/atoms/PrivateRoute/PrivateRoute.tsx:11]: It does't work at all

**Pull request merged**